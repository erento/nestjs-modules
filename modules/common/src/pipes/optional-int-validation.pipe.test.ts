import {OptionalIntValidationPipe} from './optional-int-validation.pipe';

describe('OptionalIntValidationPipe', (): void => {
    let pipe: OptionalIntValidationPipe;

    beforeEach((): void => {
        pipe = new OptionalIntValidationPipe();
    });

    test.each<[string | undefined, number | undefined]>([
        ['-1', -1],
        ['0', 0],
        ['1', 1],
        ['1234567890', 1234567890],
        ['1234.56789', 1234],
        [undefined, undefined],
    ])(
        'should transform string to value properly',
        (input: string | undefined, expected: number | undefined): void => {
            expect(pipe.transform(input))
                .toEqual(expected);
        },
    );
    test('should fail on empty string', (): void => {
        expect((): number | undefined => pipe.transform(''))
            .toThrowErrorMatchingSnapshot();
    });
    test('should fail on invalid string', (): void => {
        expect((): number | undefined => pipe.transform('asdf'))
            .toThrowErrorMatchingSnapshot();
    });
});
