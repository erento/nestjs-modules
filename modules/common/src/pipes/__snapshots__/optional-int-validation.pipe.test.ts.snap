// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionalIntValidationPipe should fail on empty string 1`] = `"Value is not a number: ''"`;

exports[`OptionalIntValidationPipe should fail on invalid string 1`] = `"Value is not a number: 'asdf'"`;
